src/pilot/deploy-overcloud.py

```diff
@@ -30,18 +37,370 @@
+            aggregate_meta = [{'pinned': 'False'}, {'hugepages': 'False'}]
+            config_oc_obj.configure_dell_aggregate(aggregate_name,
+                                                   aggregate_meta)
+    except Exception as error:
+        message = "Exception {}: {}".format(type(error).__name__, str(error))
+        logger.error("{}".format(message))
+
+
+def get_dell_compute_nodes_uuids():
+    logger.info("Getting dell compute node uuids")
+    # Get the undercloud details
+    os_auth_url, os_tenant_name, os_username, os_password = \
+        CredentialHelper.get_undercloud_creds()
+
+    kwargs = {'username': os_username,
+              'password': os_password,
+              'auth_url': os_auth_url,
+              'project_id': os_tenant_name}
+
+    # Create nova client object
+    nova = nova_client.Client(2, **kwargs)
+
+    # Create the ConfigOvercloud object
+    config_oc_obj = ConfigOvercloud(nova)
+    dell_compute_nodes_uuid = config_oc_obj.get_dell_compute_nodes_uuid()
+    logger.debug("dell_compute_nodes_uuid: {}".format(dell_compute_nodes_uuid))
+    return dell_compute_nodes_uuid
+
+
+def reboot_compute_nodes():
+    logger.info("Rebooting dell compute nodes")
+    # Get the undercloud details
+    os_auth_url, os_tenant_name, os_username, os_password = \
+        CredentialHelper.get_undercloud_creds()
+
+    kwargs = {'username': os_username,
+              'password': os_password,
+              'auth_url': os_auth_url,
+              'project_id': os_tenant_name}
+
+    n_client = nova_client.Client(2, **kwargs)
+
+    dell_compute = []
+    for compute in n_client.servers.list(detailed=False,
+                                         search_opts={'name': 'compute'}):
+        dell_compute.append(n_client.servers.ips(compute)['ctlplane'][0]['addr'])
+
+        logger.info("Rebooting server with id: ".format(compute_id))
+        n_client.servers.reboot(compute.id)
+
+        retries = 32
+        ssh_success_count = 0
+
+        # Wait for 30s for the nodes to be powered cycle then do the checks
+        sleep(30)
+
+        for ip in dell_compute:
+            while retries > 0:
+                exit_code, _, std_err = Ssh.execute_command(ip,
+                                                            "pwd",
+                                                            user="heat-admin",
+                                                            )
+                retries -= 1
+                if exit_code != 0:
+                    logger.info("Server with IP: {} is not responsive".format(ip))
+                else:
+                    logger.info("Server with IP: {} is responsive".format(ip))
+                    ssh_success_count += 1
+                    break
+
+                # Waiting 10s before the next attempt
+                sleep(10)
+
+        if ssh_success_count == len(dell_compute):
+            logger.info("All compute nodes are now responsive. Continuing...")
```

Comment:
Should add an "else" clause to the above "if" statement so that it logs an error and aborts if not all of the compute nodes come back up.

Reply:
else statement added.